drivers/nrf24l01p/gnrc_netif_nrf24l01p.c

```diff
@@ -0,0 +1,221 @@
+{
+    /* get upper size limit */
+    int upper_frame_len = netif->dev->driver->recv(netif->dev, NULL, 0, NULL);
+
+    uint8_t frame_buffer[upper_frame_len];
+    /* copy the payload into the packet buffer */
+    int frame_len = netif->dev->driver->recv(netif->dev, frame_buffer,
+                                                 upper_frame_len, NULL);
+    if(frame_len <= 0) {
+        return NULL;
+    }
+
+    /* allocate space for the packet in the pktbuf */
+    gnrc_pktsnip_t *frame = gnrc_pktbuf_add(NULL, NULL, frame_len,
+                                            NRF24L01P_UPPER_LAYER_PROTOCOL);
+    if (!frame) {
+        DEBUG("[nrf24l01p] recv: unable to allocate space in the pktbuf\n");
+        return NULL;
+    }
+    memcpy(frame->data, frame_buffer, frame_len);
+    return frame;
+}
+
+/**
+ * @brief   Receives a @ref net_gnrc_pkt "packet" from the network interface
+ *
+ * @pre `netif != NULL`
+ *
+ * @note The function takes the bytes received via netdev_driver_t::recv()
+ *       from gnrc_netif_t::dev and re-formats it to a
+ *       @ref net_gnrc_pkt "packet" containing a @ref net_gnrc_netif_hdr
+ *       and a payload header in receive order.
+ *
+ * @param[in] netif The network interface.
+ *
+ * @return  The packet received. Contains the payload (with the type marked
+ *          accordingly) and a @ref net_gnrc_netif_hdr in receive order.
+ * @return  NULL, if @ref net_gnrc_pktbuf was full.
+ */
+static gnrc_pktsnip_t *nrf24l01p_adpt_recv(gnrc_netif_t *netif)
+{
+    netdev_t *dev = netif->dev;
+    uint8_t header_len = 0;
+    gnrc_pktsnip_t *frame;
+    gnrc_pktsnip_t *hdr;
+    gnrc_netif_hdr_t *netif_hdr;
+    shockburst_hdr_t sb_hdr;
+
+    assert(dev);
+
+    if (!(frame = _nrf24l01p_pkt_recv(netif))) {
+        return NULL;
+    }
+    uint8_t bcast_addr[] = NRF24L01P_BROADCAST_ADDR;
+    sb_hdr.addr_width = ((uint8_t *)(frame->data))[header_len++];
+    uint8_t dst_addr_width = sb_hdr_get_dst_addr_width(&sb_hdr);
+    memcpy(sb_hdr.dst_addr,
+           ((uint8_t *)(frame->data)) + header_len,
+           dst_addr_width);
+    header_len += dst_addr_width;
+#ifdef NRF24L01P_CUSTOM_HEADER
+    uint8_t src_addr_width = sb_hdr_get_src_addr_width(&sb_hdr);
+    memcpy(sb_hdr.src_addr,
+           ((uint8_t *)(frame->data)) + header_len,
+           src_addr_width);
+    header_len += src_addr_width;
+#endif
+    DEBUG("[nrf24l01p] header length: %u\n", header_len);
+    hdr = gnrc_pktbuf_mark(frame, header_len, GNRC_NETTYPE_UNDEF);
+    if (!hdr) {
+        DEBUG("[nrf24l01p] recv: unable to mark nrf24l01p header snip\n");
+        gnrc_pktbuf_release(frame);
+        return NULL;
+    }
+    gnrc_pktbuf_remove_snip(frame, hdr);
+#ifdef NRF24L01P_CUSTOM_HEADER
+    hdr = gnrc_netif_hdr_build(sb_hdr.src_addr, src_addr_width, sb_hdr.dst_addr,
+                               dst_addr_width);
+#else
+    hdr = gnrc_netif_hdr_build(NULL, 0, sb_hdr.dst_addr, dst_addr_width);
+#endif
```

Comment:
And this could become

```C
    if (IS_USED(NRF24L01P_CUSTOM_HEADER)) {
        hdr = gnrc_netif_hdr_build(sb_hdr.src_addr, src_addr_width, sb_hdr.dst_addr,
                                   dst_addr_width);
    }
    else {
        hdr = gnrc_netif_hdr_build(NULL, 0, sb_hdr.dst_addr, dst_addr_width)
    }
```

Reply:
Ok